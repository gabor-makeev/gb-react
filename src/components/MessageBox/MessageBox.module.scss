.message-box {
  flex: 0 1 300px;
  width: 300px;
  border: 1px solid black;
  box-shadow: rgba(0, 0, 0, 0.35) 0 5px 15px;
  border-radius: 15px;
  padding: 20px;

  &__list {
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: flex-end;
    gap: 10px;
    list-style-type: none;
  }

  &__item {
    position: relative;
    text-align: right;
    padding: 15px 15px;
    border-radius: 20px;

    &__author {
      position: absolute;
      left: 15px;
      top: 5px;
      font-size: 12px;
    }
  }
}