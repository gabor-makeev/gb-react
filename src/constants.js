export const AUTHORS = {
  user: 'Gabor',
  bot: 'ðŸ¤– Bot',
};

export const STYLES = {
  color: {
    primary: '#FF4F4F',
    secondary: '#31E865',
  },
};
