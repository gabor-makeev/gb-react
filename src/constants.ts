import { Authors, Messages } from './default-types';

export const api = 'https://api.spaceflightnewsapi.net/v3/articles';

export const NAVIGATION = [
  {
    id: 1,
    name: 'Home',
    path: '/',
  },
  {
    id: 2,
    name: 'Profile',
    path: '/profile',
  },
  {
    id: 3,
    name: 'Messenger',
    path: '/messenger',
  },
];

export const DUMMY_CONTENT = {
  messages: [
    {
      body: 'placerat duis ultricies lacus sed turpis tincidunt id aliquet risus',
      userEmail: Authors.USER,
      userName: Authors.USER,
    },
    {
      body: 'et molestie ac feugiat sed lectus vestibulum mattis ullamcorper velit',
      userEmail: Authors.USER,
      userName: Authors.USER,
    },
    {
      body: 'felis eget velit aliquet sagittis id consectetur purus ut faucibus',
      userEmail: Authors.USER,
      userName: Authors.USER,
    },
  ] as Messages,
};
