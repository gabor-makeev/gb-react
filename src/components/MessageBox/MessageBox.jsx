import style from './MessageBox.module.scss';
import { STYLES } from '../../constants';

import { MessageList } from './components/MessageList/MessageList';
import { Container } from '@mui/material';

export const MessageBox = ({
  messages,
  backgroundColor = STYLES.color.primary,
  borderRadius = '15px',
  padding = '20px',
}) => {
  const messageBoxStyle = {
    backgroundColor: backgroundColor,
    borderRadius: borderRadius,
    flex: '0 1 300px',
    maxWidth: '300px',
    border: '1px solid black',
    boxShadow: 'rgba(0, 0, 0, 0.35) 0 5px 15px',
    padding: padding,
  };

  return (
    <Container
      className={style['message-box']}
      sx={messageBoxStyle}
      data-testid={'messageBox'}
      maxWidth={messageBoxStyle.maxWidth}
    >
      <MessageList messages={messages} />
    </Container>
  );
};
