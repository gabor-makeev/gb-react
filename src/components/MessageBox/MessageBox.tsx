import style from './MessageBox.module.scss';
import { STYLES } from '../../constants';

import { MessageList } from './components/MessageList/MessageList';
import { Container } from '@mui/material';
import { MessageItem } from 'src/default-types';
import { FC } from 'react';

interface MessageBoxProps {
  messages: MessageItem[];
  backgroundColor?: string;
  borderRadius?: string;
  padding?: string;
}

export const MessageBox: FC<MessageBoxProps> = ({
  messages,
  backgroundColor = STYLES.color.primary,
  borderRadius = '15px',
  padding = '20px',
}) => {
  const messageBoxStyle = {
    backgroundColor: backgroundColor,
    borderRadius: borderRadius,
    flex: '0 1 300px',
    maxWidth: '300px',
    border: '1px solid black',
    boxShadow: 'rgba(0, 0, 0, 0.35) 0 5px 15px',
    padding: padding,
    marginLeft: 'unset',
  };

  return (
    <Container
      className={style['message-box']}
      sx={messageBoxStyle}
      data-testid={'messageBox'}
      maxWidth={false}
    >
      <MessageList messages={messages} />
    </Container>
  );
};
