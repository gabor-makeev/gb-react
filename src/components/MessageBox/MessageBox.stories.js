import { MessageBox } from './MessageBox';
import { DUMMY_CONTENT, STYLES } from '../../constants';

export default {
  title: 'MessageBox',
  component: MessageBox,
};

const Template = (args) => (
  <MessageBox messages={DUMMY_CONTENT.messages} {...args} />
);

export const Primary = Template.bind({});

export const RightAngles = Template.bind({});
RightAngles.args = {
  messageBoxStyle: {
    backgroundColor: STYLES.color.primary,
    borderRadius: '0px',
  },
};

export const BoldPadding = Template.bind({});
BoldPadding.args = {
  messageBoxStyle: {
    backgroundColor: STYLES.color.primary,
    padding: '50px',
  },
};
