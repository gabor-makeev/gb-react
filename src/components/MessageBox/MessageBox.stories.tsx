import { MessageBox } from './MessageBox';
import { DUMMY_CONTENT, STYLES } from '../../constants';
import { ComponentMeta, ComponentStory } from '@storybook/react';

export default {
  title: 'MessageBox',
  component: MessageBox,
} as ComponentMeta<typeof MessageBox>;

const Template: ComponentStory<typeof MessageBox> = (args) => (
  <MessageBox {...args} messages={DUMMY_CONTENT.messages} />
);

export const Primary = Template.bind({});

export const RightAngles = Template.bind({});
RightAngles.args = {
  backgroundColor: STYLES.color.primary,
  borderRadius: '0px',
};

export const BoldPadding = Template.bind({});
BoldPadding.args = {
  backgroundColor: STYLES.color.primary,
  padding: '50px',
};
