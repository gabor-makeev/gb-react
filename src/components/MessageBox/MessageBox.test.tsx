import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { MessageBox } from './MessageBox';
import { DUMMY_CONTENT, STYLES } from '../../constants';

describe('MessageBox', () => {
  it('should render', () => {
    render(<MessageBox messages={DUMMY_CONTENT.messages} />);
  });

  it('should render with snapshot', () => {
    const { asFragment } = render(
      <MessageBox messages={DUMMY_CONTENT.messages} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with default style from messageBoxStyle prop', () => {
    render(<MessageBox messages={DUMMY_CONTENT.messages} />);

    expect(screen.getByTestId('messageBox')).toHaveStyle(
      `background-color: ${STYLES.color.primary}`
    );
  });

  it('should render with custom border radius style from borderRadius prop', () => {
    render(
      <MessageBox messages={DUMMY_CONTENT.messages} borderRadius={'100px'} />
    );

    expect(screen.getByTestId('messageBox')).toHaveStyle(
      'border-radius: 100px'
    );
  });
});
