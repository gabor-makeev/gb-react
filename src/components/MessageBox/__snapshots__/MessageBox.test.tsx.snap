// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox should render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root message-box css-ymhums-MuiContainer-root"
    data-testid="messageBox"
  >
    <ul
      class="MuiList-root MuiList-padding message_list css-v2fr7u-MuiList-root"
      data-testid="messageList"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding message css-1s94d1u-MuiListItem-root"
      >
        placerat duis ultricies lacus sed turpis tincidunt id aliquet risus
        <span
          class="message__author-sign"
        >
          Gabor
        </span>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding message css-1s94d1u-MuiListItem-root"
      >
        et molestie ac feugiat sed lectus vestibulum mattis ullamcorper velit
        <span
          class="message__author-sign"
        >
          Gabor
        </span>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding message css-1s94d1u-MuiListItem-root"
      >
        felis eget velit aliquet sagittis id consectetur purus ut faucibus
        <span
          class="message__author-sign"
        >
          Gabor
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
